mongocrypt/build.gradle.kts

```diff
@@ -0,0 +1,199 @@
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+    mustRunAfter(tasks.named("unzipJava"))
+}
+
+tasks.register<Copy>("unzipJava") {
+    /*
+        Clean up the directory first if the task is not UP-TO-DATE.
+        This can happen if the download revision has been changed and the archive is downloaded again.
+     */
+    doFirst {
+        println("Cleaning up $jnaResourcesDir")
+        delete(fileTree(jnaResourcesDir).matching {
+            include(jnaMapping.keys.flatMap {
+                listOf(
+                    "${it}/nocrypto/**/libmongocrypt.so",
+                    "${it}/lib/**/libmongocrypt.dylib",
+                    "${it}/bin/**/mongocrypt.dll"
+                )
+            })
+        })
+    }
+    from(tarTree(resources.gzip("${jnaDownloadsDir}/$localBinariesArchiveName")))
+    include(jnaMapping.keys.flatMap {
+        listOf("${it}/nocrypto/**/libmongocrypt.so", "${it}/lib/**/libmongocrypt.dylib", "${it}/bin/**/mongocrypt.dll" )
+    })
+    eachFile {
+        path = "${jnaMapping[path.substringBefore("/")]}/${name}"
+    }
+    into(jnaResourcesDir)
+    mustRunAfter("downloadJava")
+
+    doLast {
+        println("jna.library.path contents: \n  ${fileTree(jnaResourcesDir).files.joinToString(",\n  ")}")
+    }
+}
+
+tasks.register("downloadJnaLibs") {
+    dependsOn("downloadJava", "unzipJava")
+}
+
+tasks.test {
+    systemProperty("jna.debug_load", "true")
+    systemProperty("jna.library.path", jnaResources)
+    useJUnitPlatform()
+    testLogging {
+        events("passed", "skipped", "failed")
+    }
+
+    doFirst {
+        println("jna.library.path contents:")
+        println(fileTree(jnaResources)  {
+            this.setIncludes(listOf("*.*"))
+        }.files.joinToString(",\n  ", "  "))
+    }
+    dependsOn("downloadJnaLibs", "downloadJava", "unzipJava")
+}
+
+tasks.withType<AbstractPublishToMaven> {
+    description = """$description
+        | System properties:
+        | =================
+        |
+        | jnaLibsPath    : Custom local JNA library path for inclusion into the build (rather than downloading from s3)
+        | gitRevision    : Optional Git Revision to download the built resources for from s3.
+    """.trimMargin()
+}
+
+tasks.jar {
+    //NOTE this enables depending on the mongocrypt from driver-core
+   dependsOn("downloadJnaLibs")
+}
+
+tasks.javadoc {
+    if (JavaVersion.current().isJava9Compatible) {
+        (options as StandardJavadocDocletOptions).addBooleanOption("html5", true)
+    }
+}
+
+afterEvaluate {
+    tasks.jar {
+        manifest {
+            attributes(
+                "-exportcontents" to "com.mongodb.crypt.capi.*;-noimport:=true",
+                "Automatic-Module-Name" to "com.mongodb.crypt.capi",
+                "Import-Package" to "org.slf4j.*;resolution:=optional,org.bson.*",
+                "Bundle-Name" to "MongoCrypt",
+                "Bundle-SymbolicName" to "com.mongodb.crypt.capi",
+                "Private-Package" to ""
```

Comment:
Manifest configuration is preserved as it is from libmongocrypt, except for:

- `"Import-Package" `now uses a wildcard `"org.slf4j.*;resolution:=optional,org.bson.*"`,
- `"Build-Version" to gitVersion` is handled by [publish.gradle#L67](https://github.com/mongodb/mongo-java-driver/blob/c4af1c5fe20a88807bf4fa9c0071f6126424ade3/gradle/publish.gradle#L67).
- `"Bundle-Version" to gitVersion` is handled by [publish.gradle#L68](https://github.com/vbabanin/mongo-java-driver/blob/c4af1c5fe20a88807bf4fa9c0071f6126424ade3/gradle/publish.gradle#L68).